From e4a065d9af48f56d0317f363bcddd6d0b12ed372 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:27:05 -0400
Subject: [alsa] hda: remove PCI dependency in Kconfig

Message-id: <140972823573-220-patchset-jkysela@redhat.com>
Patchwork-id: 90920
O-Subject: [RHEL 7.1 PATCH 219/442] ALSA: hda - remove PCI dependency in Kconfig
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Remove the dependency on CONFIG_PCI for building hda codec drivers so
that platforms with HDA attach via means other than PCI can use them.
This was as suggested by tiwai.

Signed-off-by: Dylan Reid <dgreid@chromium.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 78e34f34ac27c9c25c6db58d05129994bed9ec25)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/Kconfig b/sound/pci/Kconfig
index daac7c7..9751429 100644
--- a/sound/pci/Kconfig
+++ b/sound/pci/Kconfig
@@ -575,8 +575,6 @@ config SND_FM801_TEA575X_BOOL
    FM801 chip with a TEA5757 tuner (MediaForte SF256-PCS, SF256-PCP and
    SF64-PCR) into the snd-fm801 driver.
 
-source "sound/pci/hda/Kconfig"
-
 config SND_HDSP
  tristate "RME Hammerfall DSP Audio"
  select FW_LOADER
@@ -886,3 +884,5 @@ config SND_YMFPCI
    will be called snd-ymfpci.
 
 endif # SND_PCI
+
+source "sound/pci/hda/Kconfig"
diff --git a/sound/pci/hda/Kconfig b/sound/pci/hda/Kconfig
index f2032dd..ac17c3f 100644
--- a/sound/pci/hda/Kconfig
+++ b/sound/pci/hda/Kconfig
@@ -1,8 +1,15 @@
-menuconfig SND_HDA_INTEL
- tristate "Intel HD Audio"
+menu "HD-Audio"
+
+config SND_HDA
+ tristate
  select SND_PCM
  select SND_VMASTER
  select SND_KCTL_JACK
+
+config SND_HDA_INTEL
+ tristate "HD Audio PCI"
+ depends on SND_PCI
+ select SND_HDA
  help
    Say Y here to include support for Intel "High Definition
    Audio" (Azalia) and its compatible devices.
@@ -13,7 +20,7 @@ menuconfig SND_HDA_INTEL
    To compile this driver as a module, choose M here: the module
    will be called snd-hda-intel.
 
-if SND_HDA_INTEL
+if SND_HDA
 
 config SND_HDA_DSP_LOADER
  bool
@@ -49,7 +56,7 @@ config SND_HDA_RECONFIG
 
 config SND_HDA_INPUT_BEEP
  bool "Support digital beep via input layer"
- depends on INPUT=y || INPUT=SND_HDA_INTEL
+ depends on INPUT=y || INPUT=SND_HDA
  help
    Say Y here to build a digital beep interface for HD-audio
    driver. This interface is used to generate digital beeps.
@@ -90,7 +97,7 @@ config SND_HDA_CODEC_REALTEK
    snd-hda-intel driver, such as ALC880.
 
 comment "Set to Y if you want auto-loading the codec driver"
- depends on SND_HDA_INTEL=y && SND_HDA_CODEC_REALTEK=m
+ depends on SND_HDA=y && SND_HDA_CODEC_REALTEK=m
 
 config SND_HDA_CODEC_ANALOG
  tristate "Build Analog Device HD-audio codec support"
@@ -100,7 +107,7 @@ config SND_HDA_CODEC_ANALOG
    snd-hda-intel driver, such as AD1986A.
 
 comment "Set to Y if you want auto-loading the codec driver"
- depends on SND_HDA_INTEL=y && SND_HDA_CODEC_ANALOG=m
+ depends on SND_HDA=y && SND_HDA_CODEC_ANALOG=m
 
 config SND_HDA_CODEC_SIGMATEL
  tristate "Build IDT/Sigmatel HD-audio codec support"
@@ -110,7 +117,7 @@ config SND_HDA_CODEC_SIGMATEL
    snd-hda-intel driver, such as STAC9200.
 
 comment "Set to Y if you want auto-loading the codec driver"
- depends on SND_HDA_INTEL=y && SND_HDA_CODEC_SIGMATEL=m
+ depends on SND_HDA=y && SND_HDA_CODEC_SIGMATEL=m
 
 config SND_HDA_CODEC_VIA
  tristate "Build VIA HD-audio codec support"
@@ -120,7 +127,7 @@ config SND_HDA_CODEC_VIA
    snd-hda-intel driver, such as VT1708.
 
 comment "Set to Y if you want auto-loading the codec driver"
- depends on SND_HDA_INTEL=y && SND_HDA_CODEC_VIA=m
+ depends on SND_HDA=y && SND_HDA_CODEC_VIA=m
 
 config SND_HDA_CODEC_HDMI
  tristate "Build HDMI/DisplayPort HD-audio codec support"
@@ -130,7 +137,7 @@ config SND_HDA_CODEC_HDMI
    Intel and Nvidia HDMI/DisplayPort codecs.
 
 comment "Set to Y if you want auto-loading the codec driver"
- depends on SND_HDA_INTEL=y && SND_HDA_CODEC_HDMI=m
+ depends on SND_HDA=y && SND_HDA_CODEC_HDMI=m
 
 config SND_HDA_I915
  bool
@@ -145,7 +152,7 @@ config SND_HDA_CODEC_CIRRUS
    snd-hda-intel driver, such as CS4206.
 
 comment "Set to Y if you want auto-loading the codec driver"
- depends on SND_HDA_INTEL=y && SND_HDA_CODEC_CIRRUS=m
+ depends on SND_HDA=y && SND_HDA_CODEC_CIRRUS=m
 
 config SND_HDA_CODEC_CONEXANT
  tristate "Build Conexant HD-audio codec support"
@@ -155,7 +162,7 @@ config SND_HDA_CODEC_CONEXANT
    snd-hda-intel driver, such as CX20549.
 
 comment "Set to Y if you want auto-loading the codec driver"
- depends on SND_HDA_INTEL=y && SND_HDA_CODEC_CONEXANT=m
+ depends on SND_HDA=y && SND_HDA_CODEC_CONEXANT=m
 
 config SND_HDA_CODEC_CA0110
  tristate "Build Creative CA0110-IBG codec support"
@@ -165,7 +172,7 @@ config SND_HDA_CODEC_CA0110
    snd-hda-intel driver, found on some Creative X-Fi cards.
 
 comment "Set to Y if you want auto-loading the codec driver"
- depends on SND_HDA_INTEL=y && SND_HDA_CODEC_CA0110=m
+ depends on SND_HDA=y && SND_HDA_CODEC_CA0110=m
 
 config SND_HDA_CODEC_CA0132
  tristate "Build Creative CA0132 codec support"
@@ -174,7 +181,7 @@ config SND_HDA_CODEC_CA0132
    snd-hda-intel driver.
 
 comment "Set to Y if you want auto-loading the codec driver"
- depends on SND_HDA_INTEL=y && SND_HDA_CODEC_CA0132=m
+ depends on SND_HDA=y && SND_HDA_CODEC_CA0132=m
 
 config SND_HDA_CODEC_CA0132_DSP
  bool "Support new DSP code for CA0132 codec"
@@ -196,7 +203,7 @@ config SND_HDA_CODEC_CMEDIA
    snd-hda-intel driver, such as CMI9880.
 
 comment "Set to Y if you want auto-loading the codec driver"
- depends on SND_HDA_INTEL=y && SND_HDA_CODEC_CMEDIA=m
+ depends on SND_HDA=y && SND_HDA_CODEC_CMEDIA=m
 
 config SND_HDA_CODEC_SI3054
  tristate "Build Silicon Labs 3054 HD-modem codec support"
@@ -205,7 +212,7 @@ config SND_HDA_CODEC_SI3054
    (and compatibles) support in snd-hda-intel driver.
 
 comment "Set to Y if you want auto-loading the codec driver"
- depends on SND_HDA_INTEL=y && SND_HDA_CODEC_SI3054=m
+ depends on SND_HDA=y && SND_HDA_CODEC_SI3054=m
 
 config SND_HDA_GENERIC
  tristate "Enable generic HD-audio codec parser"
@@ -214,7 +221,7 @@ config SND_HDA_GENERIC
    in snd-hda-intel driver.
 
 comment "Set to Y if you want auto-loading the codec driver"
- depends on SND_HDA_INTEL=y && SND_HDA_GENERIC=m
+ depends on SND_HDA=y && SND_HDA_GENERIC=m
 
 config SND_HDA_POWER_SAVE_DEFAULT
  int "Default time-out for HD-audio power-save mode"
@@ -225,3 +232,5 @@ config SND_HDA_POWER_SAVE_DEFAULT
    power-save mode.  0 means to disable the power-save mode.
 
 endif
+
+endmenu
diff --git a/sound/pci/hda/Makefile b/sound/pci/hda/Makefile
index 5deef4f..d0d0c19 100644
--- a/sound/pci/hda/Makefile
+++ b/sound/pci/hda/Makefile
@@ -1,4 +1,5 @@
-snd-hda-intel-objs := hda_controller.o hda_intel.o
+snd-hda-intel-objs := hda_intel.o
+snd-hda-controller-objs := hda_controller.o
 # for haswell power well
 snd-hda-intel-$(CONFIG_SND_HDA_I915) += hda_i915.o
 
@@ -25,7 +26,8 @@ snd-hda-codec-via-objs := patch_via.o
 snd-hda-codec-hdmi-objs := patch_hdmi.o hda_eld.o
 
 # common driver
-obj-$(CONFIG_SND_HDA_INTEL) := snd-hda-codec.o
+obj-$(CONFIG_SND_HDA) := snd-hda-codec.o
+obj-$(CONFIG_SND_HDA) += snd-hda-controller.o
 
 # codec drivers
 obj-$(CONFIG_SND_HDA_GENERIC) += snd-hda-codec-generic.o
-- 
1.7.1